File: portlet/space/src/main/java/social/portal/webui/component/space/UISpaceMenuPortlet.java
/*
 * Copyright (C) 2003-2007 eXo Platform SAS.
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, see<http://www.gnu.org/licenses/>.
 */
package social.portal.webui.component.space;

import org.exoplatform.webui.config.annotation.ComponentConfig;
import org.exoplatform.webui.core.UIPortletApplication;
import org.exoplatform.webui.core.lifecycle.UIApplicationLifecycle;

import social.portal.webui.component.UISpaceMenu;

/**
 * {@link UISpaceMenuPortlet} used as a portlet displaying space menu. <br />
 * Created by The eXo Platform SARL
 * @author <a href="mailto:tungcnw@gmail.com">dang.tung</a>
 * @since Dec 15, 2008
 */
@ComponentConfig(
  lifecycle = UIApplicationLifecycle.class,
  template = "app:/groovy/portal/webui/space/UISpaceMenuPortlet.gtmpl"
)

public class UISpaceMenuPortlet extends UIPortletApplication {
  
  public UISpaceMenuPortlet() throws  Exception {
	  addChild(UISpaceMenu.class, null, null);
  }
}

File: web/resources/src/main/webapp/javascript/eXo/social/webui/UISpaceMenuPortlet.js
/*
 * Copyright (C) 2003-2007 eXo Platform SAS.
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation; either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, see<http://www.gnu.org/licenses/>.
 */
 
/**
 * UISpaceMenuPortlet class
 * - Show/ hide UISpaceMenuPortlet
 */
function UISpaceMenuPortlet() {
	
}

UISpaceMenuPortlet.init = function() {
	var Util = eXo.social.Util,
		shown = true,
		uiSpaceMenuPortlet = document.getElementById('UISpaceMenuPortlet'),
		uiContainerMenu = document.getElementById('UIContainer-Menu'),
		menuTdContainer = uiContainerMenu.parentNode,
		tdToggleElement = document.createElement('td');
		
	tdToggleElement.setAttribute('id', 'ToggleElement');
	tdToggleElement.setAttribute('class', 'ToggleElement');
	tdToggleElement.setAttribute('className', 'ToggleElement');
	
	Util.insertAfter(tdToggleElement, menuTdContainer);
	tdToggleElement.innerHTML = '<div class="CloseSpaceMenuPortlet"><span>&nbsp;</span></div>';
	
	//set event handler
	Util.addEventListener(tdToggleElement, 'click', function() {
		if (shown) { //hide
			uiSpaceMenuPortlet.style.display='none';
			menuTdContainer.style.width='0px';
			shown = false;
			tdToggleElement.innerHTML = '<div class="OpenSpaceMenuPortlet"><span>&nbsp;</span></div>';
		} else { //show
			uiSpaceMenuPortlet.style.display='block';
			menuTdContainer.style.width='162px';
			shown = true;
			tdToggleElement.innerHTML = '<div class="CloseSpaceMenuPortlet"><span>&nbsp;</span></div>';
		}
	});
	
}

eXo.social = eXo.social || {};
eXo.social.webui = eXo.social.webui || {};
eXo.social.webui.UISpaceMenuPortlet = UISpaceMenuPortlet;